learn/components/styling.md

```diff
@@ -3,4 +3,191 @@ title: Styling
+  - **Specificity does not matter between DOM trees.**
+    - `:host` technically has a specificity value of 0,1,0.
+    - `fancy-button` has a specificity score of 0,0,1,
+    - The latter wins even if you wrap it with `:where()` which
+      decreases it to 0,0,0.
+
+### Using `!important` to prevent outside styling
+
+The only way to completely prevent shadow tree styles from being
+overriden is to use `!important`.
+
+```html
+<style>
+  fancy-button {
+    font-size: 2rem; /* Wins */
+    color: deeppink; /* Loses */
+    font-family: serif !important; /* Loses */
+  }
+</style>
+<fancy-button>
+  <template shadowroot="open">
+    <style>
+      :host {
+        font-size: 1rem; /* Loses */
+        color: dodgerblue !important; /* Wins */
+        font-family: sans-serif !important; /* Wins */
+      }
+    </style>
+    <slot></slot>
+    Shadow text
+  </template>
+  Slotted Text
+</fancy-button>
+```
+
+## Slots
+
+When an element becomes a shadow host, it renders its shadow tree
+instead of its light tree. Slots can be used in a shadow tree to render
+elements from the light tree. We can further style these elements using
+the `::slotted()` pseudo-element function.
+
+- The order of precedence applies the same to slotted elements.
+- Only the slotted element is exposed for styling. You cannot use
+  combinators to select relative elements.
+- Pseudo-classes like `:first-child`, `:nth-of-type()` work, however,
+  the order of the elements is based on the light tree and not where
+  they’ve been slotted in the shadow tree.
+
+## Parts
+
+The CSS Shadow Parts API is a way to expose elements within a shadow
+tree to be styled by outside CSS.
+
+```html
+<fancy-button>
+  <template shadowroot="open">
+    <button part="the-actual-button">
+      <slot name="icon"></slot>
+      <slot></slot>
+    </button>
+  </template>
+  Fancy
+</fancy-button>
+```
+
+The inner button element can be referenced by styles declared outside of
+the shadow tree using the `::part()` pseudo-element function. The `part`
+attribute is set the same way classes are set on an element. An element
+can be multiple parts.
+
+```css
+fancy-button::part(the-actual-button) {
+  color: wheat;
+}
+```
+
+- Parts follow the order of precedence as well.
+- Similar to slots, relative combinators cannot be used to select
+  unexposed elements.
+- Descedents of parts are still subject to inheritance. So parts
+  effectively become more windows of influence for a shadow tree’s
+  styles.
+
+### Using `exportparts` to expose parts from a nested custom element
+
+Use the `exportparts` attribute on the nested custom element to expose
+its parts from the current shadow host for styling. The attribute takes
+a comma separated list of part names. Names can be reassigned to prevent
+conflict using a colon.
+
+## How to declare styles in the Shadow DOM
+
+### `<style>`
+
+### `<link rel="stylesheet">`
+
+- Maybe note that `<link rel="preload" href="/path-to-stylesheet"
```

Comment:
Worth noting (and demonstrating) that browsers will coalesce these requests so one style sheet is fetched and loaded even if specified in multiple shadowRoots.

I’m okay with advocating for specific patterns here too. I think we could speak about the trade offs and advantages, steer folks towards link or css modules.